### ISSUE TITLE
Something wrong with pypy zipping

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/conda-forge-pinning-feedstock#4190]
I need to skip pypy on a [feedstock](https://github.com/conda-forge/or-tools-feedstock) temporarily, but if I add
```diff
+  # TEMP: skipping pypy to unblock feedstock
+  skip: true  # [python_impl == "pypy"]
```

and rerender, it _does_ drop two builds, but gives cpython 3.9 the wrong impl:
```diff 
rename from .ci_support/linux_64_python3.9.____cpython.yaml
rename to .ci_support/linux_64_python3.9.____cpythonpython_implpypy.yaml
index 505be7f..74a41b3 100644
--- a/.ci_support/linux_64_python3.9.____cpython.yaml
+++ b/.ci_support/linux_64_python3.9.____cpythonpython_implpypy.yaml
@@ -36,6 +36,8 @@ pin_run_as_build:
     max_pin: x.x
 python:
 - 3.9.* *_cpython
+python_impl:
+- pypy
```

**Comment by hmaarrfk** [comment]
no joke, did you try single quotes? 

https://conda-forge.org/docs/maintainer/knowledge_base.html#pypy-builds

**Comment by h-vetinari** [comment]
> no joke, did you try single quotes?

Tried but no cigar. This used to work before btw.

**Comment by hmaarrfk** [comment]
yeah, i'm looking at hte pytorch and tensorflow builds. they use the syntax.

**Comment by hmaarrfk** [comment]
Maybe this comment can help:
https://github.com/conda-forge/scipy-feedstock/blob/main/recipe/meta.yaml#L8

**Comment by h-vetinari** [comment]
> Maybe this comment can help:
> https://github.com/conda-forge/scipy-feedstock/blob/main/recipe/meta.yaml#L8

I don't think so. This is just to ensure that `python_impl` gets found in the meta.yaml if it isn't used anywhere else, but 
```
skip: true  # [python_impl == "pypy"]
```
would already do that anyway.

**Comment by molinav** [comment]
I have also experienced this behaviour in a PR for the `basemap-feedstock`: https://github.com/conda-forge/basemap-feedstock/pull/126

@regro-cf-autotick-bot rerendered the repo successfully at the start of the PR, then I made one change (adding `build_platform` for `linux_aarch64`) and requested a rerendering again. After this last rerendering, several configuration files changed the implementation from CPython to PyPy. Until now, the only workaround for me was to revert the changes manually, but every time that a rerendering is now requested, this unexpected changes pops again.
